generator-preview-utils/wrap-text-canvas.ts

```diff
@@ -15,67 +15,41 @@ export const wrapText = function (
 
   // Lets iterate over each word
-  for (var n = 0; n < words.length; n++) {
+  for (const word of words) {
     // Create a test line, and measure it..
-    testLine += `${words[n]} `
-    let metrics = ctx.measureText(testLine)
+    let metrics = ctx.measureText(line + ' ' + word)
     let testWidth = metrics.width
-    // If the width of this test line is more than the max width
-    if (testWidth > maxWidth && n > 0) {
+    // If the width of this line is more than the max width
+    if (testWidth > maxWidth && ctx.measureText(word).width <= maxWidth) {
       // Then the line is finished, push the current line into "lineArray"
       lineArray.push([line, x, 0])
-      // Increase the line height, so a new line is started
-      // Update line and test line to use this word as the first word on the next line
-      line = `${words[n]} `
-      testLine = `${words[n]} `
+      // Update line to use this word as the first word on the next line
+      line = word
     } else {
       // If the test line is still less than the max width, then add the word to the current line
-      line += `${words[n]} `
-    }
-    // If we never reach the full max width, then there is only one line.. so push it into the lineArray so we return something
-    if (n === words.length - 1) {
-      lineArray.push([line, x, 0])
+      line += ' ' + word
     }
   }
 
-  const heightText = lineArray.length * lineHeight
-  let firstY = maxHeight / 2 - heightText / 2 + lineHeight
+  // Push the last words
+  lineArray.push([line, x, 0])
+
+  const textHeight = lineArray.length * lineHeight
+  let firstY = maxHeight / 2 - textHeight / 2 + lineHeight
   let y = firstY
-  lineArray = []
-  words = text.split(' ')
-  line = '' // This will store the text of the current line
-  testLine = '' // This will store the text when we add a word, to test if it's too long
 
-  for (var n = 0; n < words.length; n++) {
-    // Create a test line, and measure it..
-    testLine += `${words[n]} `
-    let metrics = ctx.measureText(testLine)
-    let testWidth = metrics.width
-    // If the width of this test line is more than the max width
-    if (testWidth > maxWidth && n > 0) {
-      // Then the line is finished, push the current line into "lineArray"
-      lineArray.push([line, x, y])
-      // Increase the line height, so a new line is started
-      y += lineHeight
-      // Update line and test line to use this word as the first word on the next line
-      line = `${words[n]} `
-      testLine = `${words[n]} `
-    } else {
-      // If the test line is still less than the max width, then add the word to the current line
-      line += `${words[n]} `
-    }
-    // If we never reach the full max width, then there is only one line.. so push it into the lineArray so we return something
-    if (n === words.length - 1) {
-      lineArray.push([line, x, y])
-    }
+  for (const line of lineArray) {
+    line[0]
+    line[2] = y;
+    y += lineHeight;
```

Comment:
we only need to calculate `y` based on the numbers of line.
Why recalculate all lines ?